R/dvloader.R

```diff
@@ -1,61 +1,92 @@
-#' @param file_names Study file or file_names name(s) - can be a vector of strings.
-#' This is the only required argument.
-#' @param use_wd for "use working directory" - a flag used when importing local files
-#' not on NFS - default value is FALSE
-#' @param prefer_sas if set to TRUE, imports sas7bdat files first before looking for
-#' RDS files (the opposite of default behavior)
-#' @return a list of dataframes
-#' @export
+
+#' Load Data Files
+#'
+#' This function loads data files from a specified directory or the current working directory.
+#' It supports loading both RDS and SAS7BDAT files.
+#'
+#' @param sub_dir [character(1)] Optional character string specifying a subdirectory. Default is NULL.
+#' @param file_names [character(1+)] Character vector of file names to load (without extension).
+#' @param use_wd [logical(1)] Logical indicating whether to use the current working directory. Default is FALSE.
+#' @param prefer_sas [logical(1)] Logical indicating whether to prefer SAS7BDAT files over RDS. Default is FALSE.
+#' @param print_file_paths [logical(1)] Logical indicating whether to print the directory path and file names.
+#' Default is FALSE.
+#'
+#' @return A named list of data frames, where each name corresponds to a loaded file.
+#'
 #' @examples
-#' \dontrun{
-#' test_data_path <- "../inst/extdata/"
-#' data_list <- load_data(
-#'   sub_dir = test_data_path,
-#'   file_names = "dummyads2",
-#'   use_wd = TRUE
-#' )
-#' }
-load_data <- function(sub_dir = NULL, file_names, use_wd = FALSE, prefer_sas = FALSE) {
-  if (is.null(file_names)) {
-    stop("Usage: load_data: file_names: Must supply at least one file name")
+#' # Get the current value of the RXD_DATA environment variable
+#' base_dir <- Sys.getenv("RXD_DATA")
+#'
+#' # Set the RXD_DATA environment variable to the path of the haven package
+#' Sys.setenv(RXD_DATA = find.package("haven"))
+#'
+#' data_list <- load_data(sub_dir = "examples", file_names = c("iris.sas7bdat"))
+#' str(data_list)
+#'
+#' # Reset the RXD_DATA environment variable to its original value
+#' Sys.setenv(RXD_DATA = base_dir)
+#'
+#' @export
+load_data <- function(
+    sub_dir = NULL,
+    file_names,
+    use_wd = FALSE,
+    prefer_sas = FALSE,
+    print_file_paths = FALSE) {
+  checkmate::assert_character(sub_dir, len = 1, null.ok = TRUE)
+  checkmate::assert_character(file_names, min.len = 1)
+  checkmate::assert_logical(use_wd, len = 1)
+  checkmate::assert_logical(prefer_sas, len = 1)
+
+  if (use_wd) {
+    base_dir <- getwd()
+  } else {
+    base_dir <- get_nfs_path()
   }
 
-  study_path <- "" # will be built using args
+  dir_path <- if (is.null(sub_dir)) base_dir else file.path(base_dir, sub_dir)
 
-  if (is.null(sub_dir)) {
-    study_path <- getwd()
-  } else {
-    if (use_wd) {
-      study_path <- file.path(getwd(), sub_dir)
-    } else {
-      study_path <- file.path(get_cre_path(), sub_dir)
-    }
+  file_ext <- if (prefer_sas) "sas7bdat" else "rds"
```

Comment:
This variable is not used anywhere. I think it can be removed.

Reply:
Good catch. I've removed the unused `file_ext` variable since it's now handled in the `get_file_paths()` function.